lib/service/service.go

```diff
@@ -3797,6 +3816,8 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 		return trace.Wrap(err)
 	}
 
+	process.setProxyNodeWatcher(nodeWatcher)
```

Comment:
We must call `nodeWatcher.WaitInitialization()` to prevent `refreshStaleNodes` from racing with `init` goroutine.
We had this issue in the past where the watch loop initialization happens after a call to `GetNodes` - `GetNodes` ends up calling `refreshStaleNodes` because the watcher wasn't initialized - and when the init runs, it replaces the nodes because of out-of-order requests.

https://github.com/gravitational/teleport/pull/30671